File: jvm/cucumber-blueprint-jvm-example/src/test/java/cucumber/blueprint/jvm/example/integration/cucumber/step/common/BaseStepDefinition.java
package cucumber.blueprint.jvm.example.integration.cucumber.step.common;

import cucumber.blueprint.jvm.example.infrastructure.integration.camel.ExampleOsgiService;
import cucumber.runtime.java.blueprint.BlueprintDescriptorLocation;
import org.apache.camel.CamelContext;
import org.apache.camel.ProducerTemplate;

import javax.inject.Inject;

@BlueprintDescriptorLocation(value = "/OSGI-INF/blueprint/blueprintContext.xml")
public abstract class BaseStepDefinition {

    @Inject
    protected ProducerTemplate producerTemplate;

    @Inject
    protected CamelContext camelContext;

    @Inject
    protected ExampleOsgiService exampleOsgiService;
}

File: scala/cucumber-blueprint-scala-example/src/test/scala/cucumber/blueprint/jvm/example/integration/cucumber/step/common/BaseStepDefinition.scala
package cucumber.blueprint.jvm.example.integration.cucumber.step.common

import org.apache.camel.{CamelContext, ProducerTemplate}
import cucumber.blueprint.jvm.example.infrastructure.integration.camel.ExampleOsgiService
import javax.inject.Inject
import cucumber.runtime.java.blueprint.BlueprintDescriptorLocation

@BlueprintDescriptorLocation(value = "/OSGI-INF/blueprint/blueprintContext.xml")
class BaseStepDefinition {
  @Inject
  protected var producerTemplate: ProducerTemplate = null

  @Inject
  protected var camelContext: CamelContext = null

  @Inject
  protected var exampleOsgiService: ExampleOsgiService = null

  protected var result: String = _
}
